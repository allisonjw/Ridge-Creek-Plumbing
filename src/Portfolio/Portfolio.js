import React from 'react';
import './Portfolio.scss';
import portfolioImage from '../images/Laurana_Article_Image.jpg';
import Fade from 'react-reveal/Fade';
import { Link } from 'react-router-dom';

export const Portfolio = ({ allCats, projectList }) => {

    return (
      <>
        <section className="section_portfolio">
          <Fade left>
          <div className="portfolio_banner" 
          aria-label="notification of portfolio page">
            <div className="portfolio_img" aria-label="" style={{ backgroundImage:`linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(${portfolioImage})` }}>
              <h2 className="portfolio_banner-h2">
                Our Portfolio
              </h2>
            </div>
          </div>
          </Fade>
          <p className="portfolio_description-paragraph">
            Whether working with a home owner to replace a plumbing appliance, a landlord to make tenant improvements, or a commercial contractor to install a buildingâ€™s plumbing and hydronic system, our commitment to quality of service and workmanship is why we are trusted across the South Sound region. Select a category below to see more information.
          </p>
          <ul className="portfolio_navigation-bar">
            <li 
               className="portfolio_nav-li">
                <Link 
                  activeClassName="link-is_active"
                  className="portfolio_link-normal"
                  to={"/all"}>
                    ALL
                </Link>
            </li>
            <li 
              className="portfolio_nav-li">
                <Link
                  activeClassName="link-is_active"
                  className="portfolio_link-normal"
                  to="/projects/residential">
                    RESIDENTIAL
                </Link>
            </li>
            <li 
              className="portfolio_nav-li">
                <Link
                  activeClassName="link-is_active"
                  className="portfolio_link-normal"
                  to="/projects/commercial">
                    COMMERCIAL
                </Link>
            </li>
            <li 
               className="portfolio_nav-li">
                <Link
                  activeClassName="link-is_active"
                  className="portfolio_link-normal"
                  to="/projects/mixedUse">
                    MIXED USE
                </Link>
            </li>
            <li 
               className="portfolio_nav-li">
                <Link
                  activeClassName="link-is_active"
                  className="portfolio_link-normal"
                  to="/projects/hospitality">
                    HOSPITALITY
                </Link>
            </li>
          </ul> 
        </section>
          <section className="portfolio_main">
            {allCats}
            {projectList}
          </section> 
          </>    
    )
};

export default Portfolio;