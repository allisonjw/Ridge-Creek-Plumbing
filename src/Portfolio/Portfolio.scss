@import '../SCSS/media-queries';
@import '../SCSS/variables';


.portfolio_banner {
    width: 90%;
    height: 70vh;
    margin: 2em auto;
    position: relative;
    border-bottom: 1px solid #000;
    padding-bottom: 1%;
}

.portfolio_img {
    width: 100%;
    height: 70vh;
    background-size: cover;
    background-attachment: fixed;
    background-position: 50% 50%;
    background-repeat: no-repeat;
    .portfolio_banner-h2 {
        position: absolute;
        font-size: 4em;
        color: white;
        top: 18%;
        left: 14%;
        text-shadow: 5px 4px 6px #000000b5;
    }
}

.portfolio_description-paragraph {
    color: $main_color;
    font-size: 1.4em;
    line-height: 1.7;
    text-align: center;
    font-weight: 500;
    margin: 0 5%;
    padding-bottom: 1%;
}

.portfolio_navigation-bar {
    list-style: none;
    padding: 25px 0px;
    margin-bottom: 40px;
    margin-left: auto;
    margin-right: auto;
    text-align: center;
    background-color: $accent_color;
}

.portfolio_nav-li {
    display: inline-block;
    letter-spacing: 0.1em;
    margin: 0 15px;
    font-size: 1.3em;
    font-weight: 600;
    transition: all 0.3s ease 0s;
    &:hover {
        cursor: pointer;
    }
}

a.portfolio_link-normal,
a.portfolio_link-is_active {
    color: $main_color;
    text-decoration: none;
    &:hover {
        border-bottom: 1px solid $main_color;
    }
}

a.portfolio_link-normal {
    border: none;
}

a.portfolio_link-is_active {
    color: $accent_color;
    background-color: $main_color;
    border-bottom: 1px solid pink;
}

.portfolio_main {
    display: inline-flex;
    margin-right: auto;
    flex-wrap: wrap;
}
