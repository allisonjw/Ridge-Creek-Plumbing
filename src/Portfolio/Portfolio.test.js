import React from 'react';
import Portfolio from './Portfolio';
import { shallow } from 'enzyme';

describe('Portfolio', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Portfolio />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});