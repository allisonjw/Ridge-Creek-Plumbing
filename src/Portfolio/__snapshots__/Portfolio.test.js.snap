// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio should match the snapshot 1`] = `
<section
  className="section_portfolio"
>
  <Fade
    left={true}
  >
    <div
      aria-label="notification of portfolio page"
      className="portfolio_banner"
    >
      <div
        aria-label=""
        className="portfolio_img"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(Laurana_Article_Image.jpg)",
          }
        }
      >
        <h2
          className="portfolio_banner-h2"
        >
          Our Portfolio
        </h2>
      </div>
    </div>
  </Fade>
  <p
    className="portfolio_description-paragraph"
  >
    Whether working with a home owner to replace a plumbing appliance, a landlord to make tenant improvements, or a commercial contractor to install a buildingâ€™s plumbing and hydronic system, our commitment to quality of service and workmanship is why we are trusted across the South Sound region.
    <br />
    <br />
    Prior to starting Ridgecreek Plumbing, Garth was a UA member in several different states, working under various plumbing codes. He believed in the standard of excellence established by the UA, which can be found
    <a
      className="portfolio_info-links"
      href="https://uanet.org/pdf/docs/Stand_For_Excellence.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      here.
    </a>
    He believed in the standard of safety established by the UA, which can be found
    <a
      className="portfolio_info-links"
      href="https://uanet.org/pdf/docs/Stand_For_Safety_Brochure.pdf"
      rel="noopener noreferrer"
      target="_blank"
    >
      here
    </a>
    Ridgecreek Plumbing, LLC is also a member of the Olympia Master Builders Association. A link to their website is
    <a
      className="portfolio_info-links"
      href="https://omb.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      here
    </a>
    <br />
    <br />
    Select a category below to see more information.
  </p>
  <ul
    className="portfolio_navigation-bar"
  >
    <li
      className="portfolio_nav-li"
    >
      <Link
        activeClassName="link-is_active"
        className="portfolio_link-normal"
        to="/all"
      >
        ALL
      </Link>
    </li>
    <li
      className="portfolio_nav-li"
    >
      <Link
        activeClassName="link-is_active"
        className="portfolio_link-normal"
        to="/projects/residential"
      >
        RESIDENTIAL
      </Link>
    </li>
    <li
      className="portfolio_nav-li"
    >
      <Link
        activeClassName="link-is_active"
        className="portfolio_link-normal"
        to="/projects/commercial"
      >
        COMMERCIAL
      </Link>
    </li>
    <li
      className="portfolio_nav-li"
    >
      <Link
        activeClassName="link-is_active"
        className="portfolio_link-normal"
        to="/projects/mixedUse"
      >
        MIXED USE
      </Link>
    </li>
    <li
      className="portfolio_nav-li"
    >
      <Link
        activeClassName="link-is_active"
        className="portfolio_link-normal"
        to="/projects/hospitality"
      >
        HOSPITALITY
      </Link>
    </li>
  </ul>
  <section
    className="portfolio_main"
  />
</section>
`;
